src/main/java/org/opensearch/neuralsearch/search/query/HybridQueryPhaseSearcher.java

```diff
@@ -48,24 +53,115 @@
 @Log4j2
 public class HybridQueryPhaseSearcher extends QueryPhaseSearcherWrapper {
 
+    final static int MAX_NESTED_SUBQUERY_LIMIT = 50;
+
     public HybridQueryPhaseSearcher() {
         super();
     }
 
     public boolean searchWith(
         final SearchContext searchContext,
         final ContextIndexSearcher searcher,
-        final Query query,
+        Query query,
         final LinkedList<QueryCollectorContext> collectors,
         final boolean hasFilterCollector,
         final boolean hasTimeout
     ) throws IOException {
-        if (query instanceof HybridQuery) {
+        if (isHybridQuery(query, searchContext)) {
+            query = extractHybridQuery(searchContext, query);
             return searchWithCollector(searchContext, searcher, query, collectors, hasFilterCollector, hasTimeout);
         }
+        validateQuery(query);
         return super.searchWith(searchContext, searcher, query, collectors, hasFilterCollector, hasTimeout);
     }
 
+    private boolean isHybridQuery(final Query query, final SearchContext searchContext) {
+        if (query instanceof HybridQuery) {
+            return true;
+        } else if (hasNestedFieldOrNestedDocs(query, searchContext) && mightBeWrappedHybridQuery(query)) {
+            // checking if this is a hybrid query that is wrapped into a Bool query by core Opensearch code
+            // https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/DefaultSearchContext.java#L367-L370.
+            // main reason for that is performance optimization, at time of writing we are ok with loosing on performance if that's unblocks
+            // hybrid query for indexes with nested field types.
+            // in such case we consider query a valid hybrid query. Later in the code we will extract it and execute as a main query for
+            // this search request.
+            // below is sample structure of such query:
```

Comment:
we should add a TODO here which states that, we revert the change which was done in core to extract the hybrid query, but all the subqueries of the hybrid query clause needs to be again passed through the same logic of OpenSearch to wrap the queries properly to ensure that there is no latency regression.

Reply:
ack